sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -56,29 +56,50 @@ case class CreateArray(children: Seq[Expression]) extends Expression {
   }
 
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val unsafeArrayClass = classOf[UnsafeArrayData].getName
     val arrayClass = classOf[GenericArrayData].getName
     val values = ctx.freshName("values")
     ctx.addMutableState("Object[]", values, s"this.$values = null;")
```

Comment:
This line should not be here now. `values` can be `Object[]` or `${javaDataType}[]`, depending `isPrimitiveArray`.